Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -224,37 +256,142 @@ public class HTTPClient {
+    ///     - url: The URL path and query that will be sent to the server.
+    ///     - body: Request body.
+    ///     - deadline: Point in time by which the request must complete.
+    public func put(socketPath: String, url: String, body: Body? = nil, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(socketPath: socketPath, url: url, method: .PUT, body: body, deadline: deadline)
+    }
+
+    /// Execute `PUT` request to a unix domain socket path over TLS, using the specified URL as the request to send to the server.
+    ///
+    /// - parameters:
+    ///     - secureSocketPath: The path to the unix domain socket to connect to.
+    ///     - url: The URL path and query that will be sent to the server.
+    ///     - body: Request body.
+    ///     - deadline: Point in time by which the request must complete.
+    public func put(secureSocketPath: String, url: String, body: Body? = nil, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(secureSocketPath: secureSocketPath, url: url, method: .PUT, body: body, deadline: deadline)
+    }
+
+    /// Execute `DELETE` request using specified URL.
+    ///
+    /// - parameters:
+    ///     - url: Remote URL.
+    ///     - deadline: The time when the request must have been completed by.
+    public func delete(url: String, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(url: url, method: .DELETE, deadline: deadline)
+    }
+
+    /// Execute `DELETE` request to a unix domain socket path, using the specified URL as the request to send to the server.
+    ///
+    /// - parameters:
+    ///     - socketPath: The path to the unix domain socket to connect to.
+    ///     - url: The URL path and query that will be sent to the server.
+    ///     - deadline: The time when the request must have been completed by.
+    public func delete(socketPath: String, url: String, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(socketPath: socketPath, url: url, method: .DELETE, deadline: deadline)
+    }
+
+    /// Execute `DELETE` request to a unix domain socket path over TLS, using the specified URL as the request to send to the server.
+    ///
+    /// - parameters:
+    ///     - secureSocketPath: The path to the unix domain socket to connect to.
+    ///     - url: The URL path and query that will be sent to the server.
+    ///     - deadline: The time when the request must have been completed by.
+    public func delete(secureSocketPath: String, url: String, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(secureSocketPath: secureSocketPath, url: url, method: .DELETE, deadline: deadline)
+    }
+
+    /// Execute arbitrary HTTP request using specified URL.
+    ///
+    /// - parameters:
+    ///     - url: Request url.
+    ///     - method: Request rethod.
```

Comment:
```suggestion
    ///     - method: Request method.
```

Reply:
Nw that I think about this a bit more, I'm wondering if it should be `func execute(method: HTTPMethod, url: String, body: Body? = nil, deadline: NIODeadline? = nil)` (swapping method and url)